src/redux/categories/categoriesSlice.js

```diff
@@ -0,0 +1,19 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categoriesArr: [],
+  completed: false,
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: (state) => {
+      state.completed = state.categoriesArr.length > 0;
+    },
+  },
+});
+
+export const { checkStatus } = categoriesSlice.actions;
+export default categoriesSlice.reducer;
```

Comment:
According to the project requirement, your categorySlice should include an array of categories (initial state: empty array) and a reducer that checks the status and always returns "Under construction" (the initial state should check to that string). This should look similar to the code below:
```
import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

Reply:
Hello @rotimiazeez 
Thank you for the suggestion provided, however, I was working with the project requirements, and they gave us a note saying **"NOTE: No changes in React components are required in this project."**
Also, the button is not in the wireframe provided for the project. The categories section only shows the text "Under construction". How then should I implement the button?

Reply:
Changes to the categoriesSlice.js file have been implemented.

Reply:
Hi @Bennyjoez,
Implementing the highlighted code here is file